src/hotspot/share/gc/g1/g1ConcurrentRefineStats.cpp

```diff
@@ -32,6 +32,12 @@ G1ConcurrentRefineStats::G1ConcurrentRefineStats() :
   _dirtied_cards(0)
 {}
 
+double G1ConcurrentRefineStats::refinement_rate_ms() const {
+  // Report 0 when no time recorded because no refinement performed.
+  double secs = refinement_time().seconds();
+  return (secs > 0) ? (refined_cards() / (secs * MILLIUNITS)) : 0.0;
```

Comment:
Not sure how this is used, but in similar calculations we only use and return a value if the elapsed time is large enough (something like a microsecond or so) to remove extreme outliers. Not sure if that is needed here too.

Reply:
This change only uses it for logging.  I could just drop this function and do the equivalent calculation at the point where it's used.  I don't think a made up lower limit really makes that much sense anyway; I remember some places like you describe, but I thought many had been eliminated.  I could only find a couple.  The expectation here is that refined_cards() is probably also zero, unless the time granularity is rather coarse.